vnext/Microsoft.ReactNative.Cxx/JSI/JsiAbiApi.cpp

```diff
@@ -211,34 +146,10 @@ operator()(JsiRuntime const &runtime, JsiValueRef const &thisArg, array_view<Jsi
   throw;
 }
 
-/*static*/ uint32_t JsiHostFunctionWrapper::GetNextFunctionId() noexcept {
-  return ++s_functionIdGenerator;
-}
-
-/*static*/ void JsiHostFunctionWrapper::RegisterHostFunction(
-    uint32_t functionId,
-    JsiObjectRef const &functionData) noexcept {
-  std::scoped_lock lock{s_functionMutex};
-  auto it = s_functionIdToFunctionWrapper.find(functionId);
-  VerifyElseCrashSz(it != s_functionIdToFunctionWrapper.end(), "Function Id is not found.");
-  JsiHostFunctionWrapper *functionWrapper = it->second;
-  s_functionDataToFunctionWrapper[functionData.Data] = functionWrapper;
-  functionWrapper->m_functionData = functionData;
-}
-
-/*static*/ bool JsiHostFunctionWrapper::IsHostFunction(JsiObjectRef const &functionData) noexcept {
-  std::scoped_lock lock{s_functionMutex};
-  return s_functionDataToFunctionWrapper.find(functionData.Data) != s_functionDataToFunctionWrapper.end();
-}
-
-/*static*/ HostFunctionType &JsiHostFunctionWrapper::GetHostFunction(JsiObjectRef const &functionData) noexcept {
-  std::scoped_lock lock{s_functionMutex};
-  auto it = s_functionDataToFunctionWrapper.find(functionData.Data);
-  if (it != s_functionDataToFunctionWrapper.end()) {
-    return it->second->m_hostFunction;
-  }
-
-  VerifyElseCrashSz(false, "Function is not a host function");
+/*static*/ HostFunctionType &JsiHostFunctionWrapper::GetHostFunction(JsiHostFunction const &hostFunction) noexcept {
+  void *hostFunctonAbi = get_abi(hostFunction);
+  JsiHostFunctionWrapper *self = static_cast<impl::delegate<JsiHostFunction, JsiHostFunctionWrapper> *>(hostFunctonAbi);
```

Comment:
can we avoid dropping to abi? use static_pointer_cast instead?

Reply:
No, we cannot avoid dropping to ABI because the JsiHostFunction is a smart pointer to the winrt::impl::abi<JsiHostFunction> interface. We must get the interface first using the get_abi function and only after that use the static_cast on the object referenced by the JsiHostFunction.

---
In reply to: [537224452](https://github.com/microsoft/react-native-windows/pull/6693#discussion_r537224452) [](ancestors = 537224452)

Reply:
is that not what `get_self<T>` would do?
also nit: typo in the variable name hostFunct**i**onAbi :)

Reply:
Sorry, get_self does not work with delegates. Thank you for the typo catch!

---
In reply to: [537853267](https://github.com/microsoft/react-native-windows/pull/6693#discussion_r537853267) [](ancestors = 537853267)